experimental/dds/tree2/docs/Shared Tree Philosophy.md

```diff
@@ -0,0 +1,204 @@
+which applications can opt into in their schema in a compatible way.
+
+Shared Tree is also designed to ensure applications using it can also adopt this same compatibility approach while minimizing the frequency and difficulty of doing so.
+This has the largest impact around application schema evolution (changing the schema the application uses to view and edit documents).
+
+## Bipartite Tree and Field Kinds
+
+The Tree is made up of Nodes and Fields:
+
+1. Nodes are (shallowly) immutable and have a logical identity, a type and either a value or fields under string keys.
+2. Fields are collaboratively editable collections of nodes.
+
+When users edit documents, those high level semantic operations need to be encoded in a way that supports merges.
+Additionally this encoding needs to be deterministically applied in all clients, even if they are using different version of Shared Tree.
+This needs to be done such that applications can ensure invariants hold in their data model.
+This means the editing and merge logic must be kept identical for compatibility, and changes to them instead must only add support for new cases.
+For example, new edit operations can be added or new configuration flags added to existing ones, but how a particular merge gets resolved cannot be changed without introducing a compatibility issue.
+This can get complicated, so to keep it manageable, this problem was subdivided in a few different ways.
+
+The data-model for Shared Tree was selected to assist with subdividing this problem.
+The selected data-model consists of many small structures where collaborative edits interact—fields—, and nodes which connect them into a hierarchy and assigns types to each of the fields.
+
+The nodes have schema which define the schema for their fields.
+There are a few [kinds](<https://en.wikipedia.org/wiki/Kind_(type_theory)>) of nodes that configure their fields differently, for example structs (with a fixed set of mixed field types) or maps (with an extensible set of matching field types).
+
+There are also a few kinds of fields which provide different collaborative data-structures (like `sequence`, or `optional`), each of which hold nodes.
+
+All of the editing is done in the fields via the `field kind`.
+This allows the editing logic, both API and merge policy, to be packaged together into minimal versionable units ( `field kinds`).
+Since the schema explicitly selects what field kinds to use, updating them is opt-in, and can be coordinated with deployment schedules to achieve the required cross version collaboration requirements.
+This also leverages all the same collaboration as the rest of schema evolution, making adopting alternative field kinds no different from updating other aspects of the application's schema.
+
+This results in a tree where the overall shape is controlled by the application via its tree schema (for the nodes) and the alternating layers of fields provide all the complex logic, neatly separating the concerns.
```

Comment:
This is the first time I see this explanation of nodes vs fields. I think it needs more prominence as we create the final versions of the docs.